src/Xamarin.Android.Build.Tasks/Utilities/ManagedResourceParser.cs

```diff
@@ -198,35 +204,42 @@ public CodeTypeDeclaration Parse (string resourceDirectory, IEnumerable<string>
 			}
 
 			var sb = new StringBuilder ();
-			SortedDictionary<int, CodeMemberField> arrayValues = new SortedDictionary<int, CodeMemberField> ();
+
+			List<(int Key, CodeMemberField Value)> arrayValues = new List<(int Key, CodeMemberField Value)> ();
 			int value;
+			Log.LogDebugMessage ($"Processing styleable array values");
 			foreach (var kvp in arrayMapping) {
 				CodeMemberField field = kvp.Key;
 				CodeArrayCreateExpression expression = field.InitExpression as CodeArrayCreateExpression;
 				CodeMemberField [] fields = kvp.Value;
-
 				int count = expression.Initializers.Count;
 				sb.Clear ();
 				arrayValues.Clear ();
+				Log.LogDebugMessage ($"Processing array values for {field.Name} count = {count}");
 				for (int i = 0; i < count ; i++) {
 					string name = fields [i].Name;
+					Log.LogDebugMessage ($"Processing array entry {name}");
 					if (name.StartsWith ("android:", StringComparison.OrdinalIgnoreCase)) {
 						name = name.Replace ("android:", string.Empty);
 						var element = publicXml?.XPathSelectElement ($"/resources/public[@name='{name}']") ?? null;
 						value = Convert.ToInt32 (element?.Attribute ("id")?.Value ?? "0x0", fromBase: 16);
-						
 					} else {
 						CodePrimitiveExpression initExpression = fields [i].InitExpression as CodePrimitiveExpression;
 						value = Convert.ToInt32 (initExpression.Value);
 					}
-					arrayValues.Add (value, fields [i]);
+					Log.LogDebugMessage ($"{name} = {value}");
+					arrayValues.Add ((Key: value, Value: fields [i]));
 				}
+				arrayValues.Sort (new CompareTuple ());
+				Log.LogDebugMessage ($"{field.Name} has {arrayValues.Count} items.");
 				int index = 0;
 				foreach (var arrayValue in arrayValues) {
 					value = arrayValue.Key;
 					CodeMemberField f = arrayValue.Value;
 					CodePrimitiveExpression code = expression.Initializers [index] as CodePrimitiveExpression;
+					Log.LogDebugMessage ($"Setting {field.Name} {index} = {value}");
 					code.Value = value;
+					Log.LogDebugMessage ($"Creating styleable {field.Name}_{f.Name}");
```

Comment:
Are these going to be too many `LogDebugMessage` calls? The task already logs quite a few lines.